contrib/upgrade_testnet/setup_genesis.py

```diff
@@ -48,8 +48,15 @@
     genesis['app_state']['gov']['params']['voting_period'] = '120s'
     genesis['app_state']['gov']['params']['min_deposit'] = [{'amount': '10000000', 'denom': 'udaric'}]
 
+    # x/ibc
+    allowed_clients = chain_state['app_state']['ibc']['client_genesis']['params']['allowed_clients']
+    if '09-localhost' not in allowed_clients:
+        allowed_clients.append('09-localhost')
+    genesis['app_state']['ibc']['client_genesis']['params']['allowed_clients'] = allowed_clients
```

Comment:
This is required otherwise the chain won't start locally. The `09-localhost` is also inside the [default allowed clients list](https://github.com/cosmos/ibc-go/blob/0dbd3f811928b216418a45ea164d184eec86cc67/modules/core/02-client/types/params.go#L12). I believe it wasn't inside previous versions.

Reply:
I checked the IBC migration document, the localhost client should be upgraded inside the handler manually, we missed it.
https://ibc.cosmos.network/main/migrations/v7-to-v7_1.html

Reply:
Thanks @dadamu, I have now merged your code into this pranch with #1216